File: java/TwiceAsOld.java
// calculate how many years from(or to) when a father is twice as old as his son

public class TwiceAsOld {

  public static int TwiceAsOld(int dadYears, int sonYears) {
    return Math.abs(dadYears - sonYears * 2);
  }

  public static void main(String[] args) {
    System.out.println(TwiceAsOld(39, 7)); // 25
    System.out.println(TwiceAsOld(39, 21)); // 3
  }
}

File: js/Twice as old.js
//Try # 2
const twiceAsOld = (dadAge, sonAge) => Math.abs(dadAge - 2 * sonAge);

/*
function twiceAsOld(dadAge, sonAge) {
  const ageDiff = dadAge - sonAge;
  let hypSon = 1;
  let hypDad = hypSon + ageDiff; 
  while (hypSon * 2 !== hypDad) {
    hypSon++;
    hypDad++;
  }
  return Math.abs(hypDad - dadAge);
}

Your function takes two arguments:

    current father's age (years)
    current age of his son (years)

Сalculate how many years ago the father was twice as old as his son (or in how many years he will be twice as old). The answer is always greater or equal to 0, no matter if it was in the past or it is in the future.
*/
